### PR TITLE
[MOS-948] Fix for tethering popup was losing user unsaved data

### PR DESCRIPTION
**Pull request by rrandomsky**
Fixed data loss when creating or editing a note/contact after connecting the USB cable and selecting the option to enable tetering in the popup window that appears. Now, when the user wants to enable tethering, they will be asked if they want to exit without saving.

**Your checklist for this pull request**

Make sure that this PR:
- [x] Complies with our guidelines for contributions
- [ ] Has unit tests if possible
- [ ] Has documentation updated
- [x] Has changelog entry added

### CODE REVIEW

File: module-apps/apps-common/ApplicationCommon.cpp [hunk 1]
```diff
@@ -1015,6 +1015,11 @@ namespace app
         return window;
     }
 
+    uint32_t ApplicationCommon::getSizeOfWindowsStack()
```
Comment (dawidwojtas-mudita) [inline module-apps/apps-common/ApplicationCommon.cpp:1018]: Use `std::uint32_t`

File: module-apps/apps-common/WindowsStack.cpp [hunk 2]
```diff
@@ -135,5 +135,9 @@ namespace app
             std::advance(it, 1);
         }
     }
+    std::uint32_t WindowsStack::getSize() const noexcept
```
Comment (dawidwojtas-mudita) [inline module-apps/apps-common/WindowsStack.cpp:138]: I assume that there should be `std::size_t` as returning a value.

**Review by dawidwojtas-mudita — APPROVED** [review]

File: module-apps/apps-common/ApplicationCommon.cpp [hunk 3]
```diff
@@ -1015,6 +1015,11 @@ namespace app
         return window;
     }
 
+    std::uint32_t ApplicationCommon::getSizeOfWindowsStack()
```
Comment (lukasmas) [inline module-apps/apps-common/ApplicationCommon.cpp:1018]: I think it also should be "std::size_t"

File: module-apps/apps-common/popups/TetheringConfirmationPopup.cpp [hunk 4]
```diff
@@ -21,12 +21,50 @@ namespace gui
...
+            for (auto windowOnStackNumber = windowStackSize; windowOnStackNumber > 0; windowOnStackNumber--) {
+                if (const auto &previousWindowName = application->getPreviousWindow(windowOnStackNumber);
+                    previousWindowName.has_value()) {
+                    const auto previousWindow = application->getWindow(previousWindowName.value());
+                    mayUnsavedData            = previousWindow->isAnyUnsavedUserDataInWindow() ? true : mayUnsavedData;
+                }
+            }
+
+            // what is the expected action when YES is selected
+            auto whatToDoOnYes = [this]() {
```
Comment (lukasmas) [inline module-apps/apps-common/popups/TetheringConfirmationPopup.cpp:36]: What about:
auto onYesAction() ?

File: module-apps/apps-common/popups/TetheringConfirmationPopup.cpp [hunk 5]
```diff
@@ -21,12 +21,50 @@ namespace gui
         metadata.text   = utils::translate("tethering_enable_question");
         metadata.icon   = "tethering_128px_W_G";
         metadata.action = [this]() {
-            application->bus.sendUnicast(std::make_shared<sys::TetheringEnabledResponse>(),
-                                         service::name::system_manager);
-            app::manager::Controller::sendAction(application, app::manager::actions::Home);
+            // check if there is any unsaved data in any window in stack
+            bool mayUnsavedData  = false;
```
Comment (lukasmas) [inline module-apps/apps-common/popups/TetheringConfirmationPopup.cpp:25]: may doesn't sound like true or false 
If this variable doesn't keep the probability or % chance that some data is unsaved consider rename it to 
IsDataUnsaved

**Review by lukasmas — APPROVED** [review]

File: module-apps/apps-common/popups/TetheringConfirmationPopup.cpp (hunk 4 above)
Comment (rrandomsky) [reply module-apps/apps-common/popups/TetheringConfirmationPopup.cpp:36]: Yes, that's a much better name, thx

**Pull request merged**